### PR TITLE
[IMP][event_sale] Cancel registrations on sale_order cancel

### PR DESCRIPTION
**Pull request by chienandalu**
Current behavior before PR:
- When a event sale order is canceled its canceled, its registrations remain on their current states.

Desired behavior after PR is merged:
- When an event sale order is canceled it is expected to have its
registrations canceled as well.

@cc Tecnativa
--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by chienandalu** [comment]
ping @sergio-teruel

File: addons/event_sale/models/sale_order.py [hunk 1]
```diff
@@ -15,6 +15,14 @@ def action_confirm(self):
             return self.env['ir.actions.act_window'].with_context(default_sale_order_id=self.id).for_xml_id('event_sale', 'action_sale_order_event_registration')
         return res
 
+    @api.multi
+    def action_cancel(self):
```
Comment (pedrobaeza) [inline addons/event_sale/models/sale_order.py]: Please put a comment about what this override makes.

**Comment by mart-e** [comment]
Hello,

Actually I do not think this is a good idea.
We usually avoid to propagate cancels too much it is a bit dangerous.
Maybe most of the time, this is what people want but not always.

What if I want to cancel the sale order because I want the attendee to come for free?
What if I have duplicated SO pointed to the same registration and want to cancel one?

Cancel is something quite manual and we should let the user do really what he wants.

**Comment by pedrobaeza** [comment]
@mart-e I think this case is different, because the link is directly involved by Odoo code: you register in the website through *website_event_sale*, and a SO is created linked to the event registration. You control through the SOs the payment status and so on. If you cancel the SO, it's because you don't want the attendee to come.

> What if I want to cancel the sale order because I want the attendee to come for free?

There's no point on keeping the registration linked to the SO in this case. We already have this case in our customer, and we strongly advise them to create a new registration for covering that.

> What if I have duplicated SO pointed to the same registration and want to cancel one?

That can't be possible through Odoo UI. Only making the trick in the DB/shell.

Please reconsider again this inclusion.

**Comment by rafaelbn** [comment]
@mart-e 

> What if I want to cancel the sale order because I want the attendee to come for free?

Odoo doesn't works like this. If you wan the attendee to come for free then you set the price to zero or 100% discount. This is the use case Odoo have in fact.

>  What if I have duplicated SO pointed to the same registration and want to cancel one?

Well, in you duplicate SO, Odoo should create a new attendee, like if you create a new one. Remember, only when you confirm the sale order Odoo start the wizard to confirm the name, phone, email from the attende...

Please talk with you colleagues from Event management, do you know how many SO are created from Website sale that the customer never finish the checkout and will not?

Ex. you have an event 50€ and you get 200 sales order from website. But only 10 of them pay, 190 will not never come or answer. 1st you have to cancel 190 sales order, and 2nd you have to cancel 190 attendees from event. Odoo doesnt filter between modes so if you want to "cancel all atenndees which sale orders are canceled" is just impossible.

**Comment by pedrobaeza** [comment]
@mart-e can you recall this with the comments we made?

**Pull request closed without merging**